substrate/primitives/state-machine/src/testing.rs

```diff
@@ -205,6 +205,22 @@ where
 		(raw_key_values, *self.backend.root())
 	}
 
+	/// Return cloned underlying raw storage key/values and the root hash.
+	///
+	/// Useful for backing up the storage in a format that can be quickly re-loaded.
+	pub fn clone_raw_snapshot(&self) -> (Vec<(Vec<u8>, (Vec<u8>, i32))>, H::Out, StateVersion) {
```

Comment:
@bkchr @ggwpez Could there be any issue with `.backend.backend_storage().clone()`? Or is there a better way to clone `TestExternalities`?

Reply:
See: https://github.com/paritytech/polkadot-sdk/pull/8756/files#r2186218833